### ISSUE TITLE
updateQuery: Can't read property data of undefined

### ISSUE THREAD

**Issue by gauravtiwari** [apollostack/react-apollo#135]
Hey,

I am trying to implement pagination using latest version of `apollo client` and `react apollo` and getting this error when merging the old props with new props in `updateQuery`. 

```js
 this.props.data.fetchMore({
   variables: { start: this.props.data.variables.start + 20 },
   updateQuery: (oldprops, { newProps }) => {
     const newPosts = newProps.data.posts;
     return {
       posts: [...oldprops.posts, ...newPosts],
     };
   },
});
```

Seems to be coming from here: https://github.com/apollostack/react-apollo/blob/master/src/graphql.tsx#L460

The error: 

![screen shot 2016-08-13 at 15 55 46](https://cloud.githubusercontent.com/assets/771039/17643875/718cef14-616e-11e6-8324-4a64a4163338.png)

**Comment by jbaxleyiii** [comment]
Ah shoot, that's right. There is a PR in the client that I needed to finish to fix that. I'll fix that bug asap and finish the PR

**Comment by gauravtiwari** [comment]
Cool. Thanks @jbaxleyiii

**Comment by jbaxleyiii** [comment]
Fixed in 0.4.2

**Comment by japrogramer** [comment]
Bug seems to be back or Am I doing something wrong?
```
function loadMore(fetchMore, path, page) {
  return fetchMore({
    variables: {
      page: page,
    },
    updateQuery(prev, { fetchMoreResult }) {
      const more = fetchMoreResult[path].edges;
      console.log(prev, fetchMoreResult)
      return Object.assign({}, prev,{
        [path]: {
          edges: [ ...prev[path].edges, ...more],
          pageInfo: fetchMoreResult[path].pageInfo,
          __typename: prev[path].__typename
        }
      })
    },
  });
}

function AuthoredPanelsPaged (props) {
  const [Page, setPage] = React.useState(1);
  const { classes, match } = props;
  const {user} = props;

  const more = (e, fetchMore) => {
    e.preventDefault();
    const newPage = Page  + 1
    const result = loadMore(fetchMore, 'byAuthoredPanelsPaged', newPage);
    setPage(newPage)
  }

  return (
      <Query
        query={ByAuthoredPanelsPagedQuery}
        variables={{ page: Page,
          uid: match.params.id ? match.params.id : '',
        }}
      >
        {({ loading, error, data, refetch, fetchMore}) => {
          if (loading) return <Linear />;
          if (error) return ``;
          if (data.byAuthoredPanelsPaged.edges.length == 0) return null;
          return (
              <Grid item>
                  <Typography variant="subheading" align='left' gutterBottom
                    onClick={(e)=> more(e, fetchMore)}>
                  <strong>View more</strong>
                  </Typography>
              </Grid>
          );
        }}
      </Query>

  );
}
```
